data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/MaxMindGeoIpProvider.java

```diff
@@ -17,18 +20,38 @@
 import java.io.IOException;
 import java.net.InetAddress;
 import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 
 
 public class MaxMindGeoIpProvider implements GeoIpProvider {
     private static final Logger LOG = LoggerFactory.getLogger(MaxMindGeoIpProvider.class);
     private final DatabaseReader databaseReader;
+    private ArrayList<Fields> fieldsToAdd = new ArrayList<>();
 
-    public MaxMindGeoIpProvider(String databasePath) {
-
+    public MaxMindGeoIpProvider(String databasePath, String[] desiredFields) {
         Objects.requireNonNull(databasePath, String.format("Missing '%s' configuration value", GeoIpPrepperConfig.DATABASE_PATH));
         File database = new File(databasePath);
+        if (desiredFields != null && desiredFields.length > 0) {
+            for (String desiredField : desiredFields) {
+                try {
+                    Fields field = Fields.valueOf(desiredField.toUpperCase(Locale.ROOT));
+                    fieldsToAdd.add(field);
+                } catch (Exception e) {
+                    LOG.error("Invalid field in config settings, {}", desiredField);
+                }
+            }
+        } else {
+            fieldsToAdd = new ArrayList<>(Arrays.asList(Fields.IP, Fields.CITY_NAME,
```

Comment:
I suggest delegate the list creation to Fields enum itself, i.e. add an allValues() method that returns the list of all values

Reply:
Actually there is built-in `values()` API in enum you could use.